src/context/WalletProvider/KeepKeyProvider.tsx

```diff
@@ -157,14 +158,31 @@ export const KeepKeyProvider = ({ children }: { children: React.ReactNode }): JS
     if (!keepKeyWallet) return
     dispatch({ type: KeepKeyActions.SET_WALLET, payload: keepKeyWallet })
     ;(async () => {
-      const features = await keepKeyWallet.getFeatures()
-      dispatch({ type: KeepKeyActions.SET_FEATURES, payload: features })
-      setHasPassphrase(features?.passphraseProtection)
-      setDeviceTimeout(
-        Object.values(timeoutOptions).find(t => Number(t.value) === features?.autoLockDelayMs),
-      )
+      await poll({
```

Comment:
tl;dr instead of it running only once and possibly too early, polls this and only runs the `features` path if polling was happy